### PR TITLE
[Security] Certificate revocation check is not done properly

### PR DESCRIPTION
**Pull request by partychen**
https://ceapex.visualstudio.com/Engineering/_workitems/edit/383462/

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/docfx/pull/7120)

### CODE REVIEW

File: src/docfx/lib/github/GitHubAccessor.cs [hunk 1]
```diff
@@ -23,6 +23,16 @@ internal sealed class GitHubAccessor
     {
         private static readonly Uri s_url = new("https://api.github.com/graphql");
 
+        private static readonly HttpClient s_sharedHttpClient =
```
Comment (live1206) [inline src/docfx/lib/github/GitHubAccessor.cs:26]: do we need to set this for all `HttpClient`?
For instance, https://github.com/dotnet/docfx/blob/be1e9e78e739e9a9d3f71eee1b96346e6178091b/src/docfx/config/ops/OpsAccessor.cs#L46

Comment (partychen) [reply src/docfx/lib/github/GitHubAccessor.cs:26]: Thanks for reminder this, find all to fix.

File: src/docfx/lib/github/GitHubAccessor.cs [hunk 2]
```diff
@@ -40,7 +50,7 @@ public GitHubAccessor(Config config)
 
             if (!string.IsNullOrEmpty(config.GithubToken))
             {
-                _httpClient = new HttpClient();
+                _httpClient = s_sharedHttpClient;
```
Comment (yufeih) [inline src/docfx/lib/github/GitHubAccessor.cs:53]: This http client cannot be shared:
```suggestion
                _httpClient = new(new HttpClientHandler { CheckCertificateRevocationList = true });
```

**Review by yufeih — CHANGES_REQUESTED** [review]

**Review by yufeih — APPROVED** [review]

**Review by live1206 — APPROVED** [review]

File: src/docfx/lib/github/GitHubAccessor.cs [hunk 3]
```diff
@@ -40,7 +40,7 @@ public GitHubAccessor(Config config)
 
             if (!string.IsNullOrEmpty(config.GithubToken))
             {
-                _httpClient = new HttpClient();
+                _httpClient = new HttpClient(new HttpClientHandler { CheckCertificateRevocationList = true });
```
Comment (yufeih) [inline src/docfx/lib/github/GitHubAccessor.cs:43]: Move the initialization to line 26 seems to fix the warning.

**Pull request merged**